### PR TITLE
ACL filter action documentation updates

### PR DESCRIPTION
**Pull request by bneradt**
This renames the ACL filter behavior as "legacy" and "modern" and
documents that the legacy behavior, while the default for 10.x, is
deprecated and will be removed in 11.x
    
This also adds support for set_allow and set_deny.

### CODE REVIEW

File: doc/admin-guide/files/remap.config.en.rst [hunk 1]
```diff
@@ -662,32 +749,63 @@ This is an example of in-line filter, named filters in :file:`remap.config`, and
...
 ====== ============== ============== ============== ================ =============
 Method In-line Filter Named Filter 1 Named Filter 2 ip_allow.yaml    result
 ====== ============== ============== ============== ================ =============
 GET    \-             \-             \-             allow (implicit) allowed (200)
-POST   deny           \-             \-             \-               denied  (403)
-HEAD   \-             allow          \-             \-               allowed (200)
-DELETE \-             \-             deny           \-               denied  (403)
+POST   deny           \-             \-             allow (implicit) denied  (403)
+HEAD   \-             allow          \-             allow (implicit) allowed (200)
+DELETE \-             \-             deny           allow (implicit) denied  (403)
```
Comment (masaori335) [inline doc/admin-guide/files/remap.config.en.rst:760]: Thanks for this clarification 👍 This table explains the above config very well.

File: doc/admin-guide/files/remap.config.en.rst [hunk 2]
```diff
@@ -662,32 +749,63 @@ This is an example of in-line filter, named filters in :file:`remap.config`, and
...
+====== ================ ============== ============== ================ =============
+Method In-line Filter   Named Filter 1 Named Filter 2 ip_allow.yaml    result
+====== ================ ============== ============== ================ =============
+GET    allow            \-             \-             allow (implicit) allowed (200)
+POST   deny             \-             \-             allow (implicit) denied  (403)
+HEAD   allow            allow          \-             allow (implicit) allowed (200)
+DELETE allow            \-             deny           allow (implicit) allowed (200)
+PURGE  allow            \-             \-             delete           allowed (200)
+PUSH   allow            \-             \-             delete           allowed (200)
+====== ================ ============== ============== ================ =============
```
Comment (masaori335) [inline doc/admin-guide/files/remap.config.en.rst:808]: We can simply say like this
```
====== ================ ============== ============== ============= =============
Method In-line Filter   Named Filter 1 Named Filter 2 ip_allow.yaml result
====== ================ ============== ============== ============= =============
GET    allow            \-             \-             allow         allowed (200)
POST   deny             \-             \-             allow         denied  (403)
HEAD   allow            allow          \-             allow         allowed (200)
DELETE allow            \-             deny           allow         allowed (200)
PURGE  allow            \-             \-             deny          allowed (200)
PUSH   allow            \-             \-             deny          allowed (200)
====== ================ ============== ============== ============= =============
```

**Review by masaori335 — CHANGES_REQUESTED** [review]
Thank you for woking this! Now, I agree with below

- ATS 10 introduce four new ACL filter actions, `set_allow`, `set_deny`, `add_allow`, and `add_deny`.
- Also, ip_allow.yaml has new actions align with above actions, `set_allow` and `set_deny`.
- When a new configuration called `proxy.config.url_remap.acl_behavior_policy` is set to `1`, ATS checks configs that have old actions (`allow` and `deny`) or not, but do nothing in default (`0`).

One last little thing is please tweak the table of "Modern" policy.

File: doc/admin-guide/files/remap.config.en.rst (hunk 2 above)
Comment (bneradt) [reply doc/admin-guide/files/remap.config.en.rst:808]: Great catch! Yes, I did mean deny. Thank you.

**Comment by bneradt** [comment]
Thank you @masaori335 for the review and all your thoughts through this process. I updated the table.

**Review by masaori335 — APPROVED** [review]
Looks good! Thanks for clarifications.

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**